src/pages/Launchpad/Summary/useSubmit/index.ts

```diff
@@ -1,86 +1,133 @@
 import { createEndowment } from "contracts/evm/Account";
-import useCosmosTxSender, { Tx } from "hooks/useCosmosTxSender";
-import { getWasmAttribute } from "helpers";
-import { estimateFee, sendTx as sendEVMTX } from "helpers/evm";
-import { EMAIL_SUPPORT, GENERIC_ERROR_MESSAGE } from "constants/common";
+import { logger } from "helpers";
+import { estimateTx, sendTx } from "helpers/tx";
+import { chainIds } from "constants/chainIds";
+import { GENERIC_ERROR_MESSAGE } from "constants/common";
 import { appRoutes } from "constants/routes";
 import toEVMAIF from "./toEVMAIF";
 import toJunoAIF from "./toJunoAIF";
 
 export default function useSubmit(network: Network) {
   const { wallet } = useGetWallet();
-  const sendTx = useCosmosTxSender();
   const [saveAIF] = useSaveAIFMutation();
   const { showModal, closeModal } = useModalContext();
   const navigate = useNavigate();
 
   async function submit(completed: Completed) {
-    if (!wallet) {
-      return showModal(TxPrompt, { error: "Wallet is not connected" });
-    }
+    try {
+      if (!wallet) {
+        return showModal(TxPrompt, { error: "Wallet is not connected" });
+      }
 
-    if (network === "polygon" /**TODO: && check connected wallet's chainID */) {
-      const tx: SimulContractTx = {
-        to: "0x09266441B8Dc93EE70Dbe27A3612eA6e1116f1F3", //TODO: move to src/contracts/evm
-        from: wallet.address,
-        data: createEndowment.encode(toEVMAIF(completed, wallet.address)),
-      };
-      const { tx: simulated } = await estimateFee(wallet, tx);
-      const hash = await sendEVMTX(wallet, simulated);
-      return alert(hash);
-    }
+      const { chain } = wallet;
+      if (
+        network === "polygon" &&
+        !(chain.chain_id === chainIds.polygon || chain.chain_id === "1337") //polygon local
+      ) {
+        /** TODO: wallet state should have `type: ("evm" | "cosmos" ..etc)` field
+         *  so the flow would be
+         *  1. Connected wallet doesn't support (non-EVM) this transaction
+         *  2. Please connect to Polygon network
+         *
+         *  if PRE-CHECK ( UI is disabled + helpful toolip)
+         *  no need to perform these checks but value must be casted
+         *  const wallet = useGetWallet() as EVM/CosmosWalletConnectedToCorrectNetwork
+         */
+        return showModal(TxPrompt, {
+          error: "Please connect to Polygon network",
+        });
+      }
 
-    if (!(wallet.providerId === "keplr" || wallet.providerId === "keplr-wc")) {
-      return showModal(TxPrompt, {
-        error: "Only Keplr wallet support this transaction",
-      });
-    }
+      if (network === "juno" && chain.chain_id !== chainIds.juno) {
+        return showModal(TxPrompt, {
+          error: "Please connect to Juno network",
+        });
+      }
+
+      // //////////////// CONSTRUCT TX CONTENT ////////////////////
+      let content: TxContent;
+      if (
+        network === "polygon" /**TODO: && check connected wallet's chainID */
```

Comment:
`/**TODO: && check connected wallet's chainID */`
Shouldn't this check also be included here?